core/commonMain/src/kotlinx/serialization/Annotations.kt

```diff
@@ -330,13 +334,15 @@ public annotation class Polymorphic
  *
  * Automatically generated serializer is available via `generatedSerializer()` function in companion object of serializable class.
  *
- * Generated serializers allow to use custom serializers on classes from which other serializable classes are inherited.
+ * Keeping generated serializers allow to use plugin generated serializer in inheritors even if custom serializer is specified.
+ *
+ * Used only with annotation [Serializable] with the specified argument [Serializable.with], e.g. `@Serializable(with=SomeSerializer::class)`.
  *
- * Used only with the [Serializable] annotation.
+ * Annotation is not allowed on classes involved in polymorphic serialization:
+ * interfaces, sealed classes, abstract classes, classes marked by [Polymorphic].
```

Comment:
I don't understand this sentence: `@KeepGeneratedSerializer` is not allowed with any inheritance, but your sample code uses a class with inherits from an interface. (see other comment)

Reply:
It means that you cannot specify the `@KeepGeneratedSerializer ` interface on the interfaces, sealed/abstract classes, but it's ok to add this annotation add to the heirs